p4-16/spec/P4-16-spec.mdk

```diff
@@ -4657,6 +4657,23 @@ table tbl {
 }
 ~ End P4Example
 
+## Operations on `extern` objects { #sec-extern-operations }
+
+The only operations that can be performed on `extern` objects are the
+following:
+
+- Instantiating an extern object using a constructor invocation, as described in
+  Section [#sec-constructor].
+- Invoking a method of an extern object instance using a method call expression, as
+  described in Section [#sec-invocations].
+
+Controls, parsers, packages, actions, and extern constructors can have
```

Comment:
There was some discussion during the 2023-Apr-10 LDWG meeting asking whether `actions` should be removed from this sentence.

After the meeting, I was reminded of the following sentence in Section 14.1 "Actions" that has been in the spec since before version 1.0 was published: "Action parameters may not have extern types. "

I traced it back as far as 2016-Dec-16 before I stopped going back in time looking further.

We can generalize this, of course, but it seems like it could use more thought on what it means, how it is defined, any restrictions on extern instances as action parameters, etc. before adding it.

Reply:
I agree -- for now the best thing would be to remove the word "actions" from the line 4670. Allowing externs to be action parameters opens a whole can of worms and while I can see how this might be useful, a closer look reveals many problems. Let's continue with the current approach (externs cannot be action parameters) -- we can always allow them later.

Reply:
Ahh, thanks.